test/Array/main.cpp

```diff
@@ -133,26 +138,37 @@ static void test_non_pod() {
     }
     for (idx = 0; idx < initial_capacity; idx ++) {
         const Test& t = array[idx];
-        MBED_HOSTTEST_ASSERT(t == Test(idx, 'b'));
+        TEST_ASSERT_TRUE(t == Test(idx, 'b'));
     }
 
     // Pop the last element from array (checks if destructor is called)
     array.pop_back();
 
-    MBED_HOSTTEST_ASSERT(array.get_num_elements() == initial_capacity - 1);
-    MBED_HOSTTEST_ASSERT(array.get_num_zones() == 1);
+    TEST_ASSERT_TRUE(array.get_num_elements() == initial_capacity - 1);
+    TEST_ASSERT_TRUE(array.get_num_zones() == 1);
+    }
+    TEST_ASSERT_TRUE(Test::inst_count == 0);
 }
 
-void app_start(int, char**) {
-    MBED_HOSTTEST_TIMEOUT(5);
-    MBED_HOSTTEST_SELECT(default);
-    MBED_HOSTTEST_DESCRIPTION(mbed-util array  test);
-    MBED_HOSTTEST_START("MBED_UTIL_ARRAY_TEST");
+static status_t test_setup(const size_t number_of_cases) {
+    GREENTEA_SETUP(5, "default_auto");
 
-    test_pod(); // test with "plain old data"
-    test_non_pod(); // test with complex data
-    MBED_HOSTTEST_ASSERT(Test::inst_count == 0);
+    return greentea_test_setup_handler(number_of_cases);
+}
+
+status_t greentea_failure_handler(const Case *const source, const failure_t reason) {
+    greentea_case_failure_abort_handler(source, reason);
+    return STATUS_CONTINUE;
+}
 
-    MBED_HOSTTEST_RESULT(true);
+static Case cases[] = {
+    Case("Array  - test with plain old data", test_pod, greentea_failure_handler),
+    Case("Array  - test with complex data", test_non_pod, greentea_failure_handler)
+};
+
+static Specification specification(test_setup, cases, greentea_test_teardown_handler);
```

Comment:
However, you don't need to define a custom `greentea_failure_handler` to make the test cases continue on failure, you can use the [`greentea_continue_handlers `](https://github.com/PrzemekWirkus/utest/blob/devel_transport/utest/default_handlers.h#L185-L186).
Also the `greentea_test_teardown_handler` is part of the [`greentea_continue_handlers`](https://github.com/PrzemekWirkus/utest/blob/devel_transport/source/greentea_handlers.cpp#L41), which is the [the default argument of the `Specification` constructor](https://github.com/PrzemekWirkus/utest/blob/devel_transport/utest/specification.h#L85-L92).

```cpp
Specification specification(test_setup, cases, greentea_continue_handlers);
```